packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java

```diff
@@ -168,25 +168,13 @@ private static String getPathFromRemoteUri(final Context context, final Uri uri)
   }
 
   /** @return extension of image with dot, or default .jpg if it none. */
-  private static String getImageExtension(Context context, Uri uriImage) {
+  private static String getImageExtension(Uri uriImage) {
     String extension = null;
-    Cursor cursor = null;
 
     try {
-      cursor =
-          context
-              .getContentResolver()
-              .query(uriImage, new String[] {MediaStore.MediaColumns.MIME_TYPE}, null, null, null);
-
-      if (cursor != null && cursor.moveToNext()) {
-        String mimeType = cursor.getString(0);
-
-        extension = MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType);
-      }
-    } finally {
-      if (cursor != null) {
-        cursor.close();
-      }
+      extension = MimeTypeMap.getFileExtensionFromUrl(uriImage.getPath());
```

Comment:
I did some research on this method and I found this: https://stackoverflow.com/questions/14320527/android-should-i-use-mimetypemap-getfileextensionfromurl-bugs/14321470
it seems getFileExtensionFromUrl is not sufficient for all scenarios?

Reply:
agreed, would be better to find the extension manually but in some scenarios, I found that file uri doesn't have an extension so in that case, it will also not sufficient. so checking for null or empty extension and applying manual extension will be working.

Reply:
Sure, do you mind to add code to manually check the extension as well? 
Something like 
```
extension = fromURI
if (extension == null) {
  extension = manuallyParsing
}
if (extension == null) {
  extension = ".jpg" (i believe this is already handled)
}

```

Reply:
did the change, null check for extension was already there, in addition to that I added an empty check to prevent rare scenarios.